distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/ErrorPumpThread.java

```diff
@@ -54,12 +54,12 @@ class ErrorPumpThread extends Thread {
      * @return a bootstrap exeption message if a bootstrap error occurred, or null otherwise
      * @throws IOException if there was a problem reading from stderr of the process
      */
-    String waitUntilReady() throws IOException {
+    boolean waitUntilReady() throws IOException {
         nonInterruptibleVoid(readyOrDead::await);
         if (ioFailure != null) {
             throw ioFailure;
```

Comment:
> Currently you have waitUntilReady rethrowing exceptions. That would not wait for the forked process to exit, nor convert that exception into a UserException as the existing code does.

Just a question regarding your point on the expected behavior on IOExceptions, currently these are rethrown as UncheckedIOException in ServerProcessBuilder. Are we good keeping it like that?